pynapple/core/time_series.py

```diff
@@ -859,6 +859,27 @@ def save(self, filename):
 
         return
 
+    @classmethod
+    def _from_npz_reader(cls, file):
+        """
+        Load a TsdTensor object from a npz file.
+
+        Parameters
+        ----------
+        file : str
+            The opened npz file
+
+        Returns
+        -------
+        Tsd
+            The Tsd object
+        """
+        return cls(
+            t=file["t"],
+            d=file["d"],
+            time_support=IntervalSet(start=file["start"], end=file["end"]),
+        )
+
```

Comment:
the code for tsdtensor and tsd is the same, and for Ts and TsFrame is almost the same. 

I would have a single class method on the super class `Base` handling all cases. The code could look something like this:


```python
    @classmethod
    def _from_npz_reader(cls, file)
        kwargs = {key: val for key, val in file.items() if key not in ["start", "end"]}
        iset = nap.IntervalSet(**{key: val for key, val in file.items() if key in ["start", "end"]})
        cls(time_support=iset, **kwargs)
```

Reply:
Makes lots of sense and yes, much more compact! I was not sure how explicit I should have been about filtering the arguments that can be found in the file. For some reason the interface class seems to assume that there can be extra content in (untested, maybe previous?) npz files. If that is not a possibility, even the code in the interface could be streamlined